### ISSUE TITLE
Imported Bones dont behave properly with looking_at

### ISSUE THREAD

**Issue by jedStevens** [godotengine/godot#17043]
No one seems to be having bone issues, which I assume is becuase no one is toying with this.

125fc8cc4432d28e63da76c3a2aff655b1a7cc6b

Manjaro

**Issue description:**
Take this script, and use it on some imported arm from blender.

This script intends to make the root bone look at the target affector, but always looks at a tangent to the desired direction.

```
export(NodePath) var target = null

func _ready():
	$skel/arm.set_surface_material(0,preload("res://skin.tres"))
	
	var bone_1 = $Skeleton.get_bone_global_pose(0)
	var target_t = $Skeleton.global_transform.affine_inverse() * get_node(target).global_transform
	bone_1.basis = bone_1.looking_at(target_t.origin, bone_1.basis[1]).basis
	$Skeleton.set_bone_global_pose(0, bone_1)
```

I expect that when using `bone_1.basis = bone_1.looking_at(target.origin, bone.basis[1])` that I am only changing the rotation and scale of the bone transform and also pointing it at the target.

Why is it that this makes the arm look at a 90 degree angle??

Does this have to do with the axis swap? Are the bones' forward facing direction accounted for in the GLTF exporter? (escn wouldn't produce a usable result) #17068 

Godot should intuitively set the bones' forward direction to -Z. This function doesn't work as expected on bones from blender. I've seen examples online about how to use a script to make bones and that returned positive results when using the `looking_at` function.

**Steps to reproduce:**
Use some imported bones and try to move them using `looking_at`, it will not point in the right direction. This is extremely frustrating as all skeletons will come from blender and this should work as intended, (ie why isn't the fucking bone pointing at the target)

**Minimal reproduction project:**
[ik.zip](https://github.com/godotengine/godot/files/1758702/ik.zip)

EDIT ---

https://blender.stackexchange.com/questions/15609/in-which-direction-should-bone-axes-be-oriented 

This answer tells that blender uses Y axis along the bone. Should this not be switched to Z- forward so we can use it like a transform?

**Comment by jedStevens** [comment]
I've spent hours trying to recalculate bone roll in blender with no usable changes to the result of `looking_at`, but when I use an object facing z- everything works fine. If I use a model facing Y+ I get the same error as blender bones produce.

**Comment by jedStevens** [comment]
Sorry to keep commenting on my own issue but im really mad at this; i've been trying to make this work for ages with my other issue just simply being archived before 3.0

What gives?

In my opinion, on the export for a bone we should set a bone's basis like so:
```
basis[0] = read the X Axis from blender
basis[2] = (tip - base).normalized()
basis[1] = cross(basis[0], basis[2])
```
I'm not sure why ANYTHING BUT Z- is considered forward in Godot. This should be converted over so we can ACTUALLY use skeletons and bones as transforms as suggested in the docs

http://docs.godotengine.org/en/stable/tutorials/3d/working_with_3d_skeletons.html

> It is a Transform, so we can do everything we can do with Transform, basically translate, rotate and scale. Also we can multiply transforms to have complex transforms. Remember, “bones” in Godot are just Transforms over a group of vertices.

**Comment by TwistedTwigleg** [comment]
I've made my own IK solver and made a [PR in the demo repository](https://github.com/godotengine/godot-demo-projects/pull/226). Maybe it will help?

I'm not sure if I had `look_at` issues or not, as it was awhile ago when I wrote the code for the look at solver. I do remember having a fair amount of trouble figuring out how to convert from bone space to world space, which was causing several problems similar to those you've mentioned (especially when the skeleton was not at origin with a rotation of `(0,0,0)` )

**Comment by jedStevens** [comment]
I wrote my own aswell and it just deals in Quaternions now instead of using `looking_at` I'm finding this works better myself. I just made a 2 bone system instead of using FABRIK however.

**Comment by reduz** [comment]
This is because of how bones work in Godot. This will be changed in 3.2

**Comment by akien-mga** [comment]
There have been changes to Skeleton and bones in 3.2, how is the current status for this issue in the latest 3.2 betas or later?

**Comment by KoBeWi** [comment]
Can anyone still reproduce this bug in Godot 3.2.3 or any later release?

**Comment by TwistedTwigleg** [comment]
I have not tested this in awhile, but I do not think its an issue per-say. The issue is that `look_at` rotates the transform so that the negative Z axis faces the target position, but a bone may have a forward direction that is not the negative Z axis (GLTF, for example, defaults to Y+ as the bone forward axis). This difference in forward axes leads to the look_at function not seeming to work correctly since the forward axes don't match up.

I worked around this issue in TwistedIK2 by rotating all bones so they face the negative Z axis for any transformations, and then undoing the rotation so they face the proper bone forward direction afterward. This allows `look_at` to be used as expected. I think I also wrote a helper function in my IK PR to take a transform with -Z as forward and convert it to the proper forward direction.

I think we can close this issue though, as it's not really a problem with the look_at function, its just bones do not always use -Z as their forward direction.
